Sources/BraveWallet/Panels/WalletPanelView.swift

```diff
@@ -511,13 +511,32 @@ struct WalletPanelView: View {
     .onChange(of: keyringStore.selectedAccount) { _ in
       isConnectHidden = isConnectButtonHidden()
     }
+    .onChange(of: tabDappStore.latestPermissionRequest) { newValue in
+      if let request = newValue, request.requestingOrigin == origin, request.coinType == keyringStore.selectedAccount.coin {
+        presentWalletWithContext(.requestEthererumPermissions(request, onPermittedAccountsUpdated: { accounts in
+          if request.coinType == .eth {
+            ethPermittedAccounts = accounts
+          } else if request.coinType == .sol {
+            solConnectedAddresses = Set(accounts)
+            isConnectHidden = false
+          }
+          tabDappStore.latestPermissionRequest = nil
+        }))
+      }
+    }
     .onAppear {
+      if let accountCreationRequest = WalletProviderAccountCreationRequestManager.shared.firstPendingRequest(for: origin, coinTypes: Array(WalletConstants.supportedCoinTypes)) {
+        presentWalletWithContext(.createAccount(accountCreationRequest))
+        return
```

Comment:
I don't think we want to `return` here, as we'll miss updating pieces of `WalletPanelView` from the bottom of this `onAppear` closure

Reply:
This is due to the question we asked James at the beginning when i started integrate this. The question is if we have account creation request and permission request at the same time, which one to display first. 
The decision is to display account creation request first, that's why we check `firstPendingRequest` for account creation request first. 
In this flow,
if user creates an solana account `tabDappStore.latestPendingPermissionRequest` will take care the flow continuity to display the permission request screen.
If user ignores it, next time they are back to wallet panel, it will prompt again 
If user decide not to create, the account creation request will be removed from the queue.

Reply:
Can't we use an else block instead of return?
```
if let accountCreationRequest = WalletProviderAccountCreationRequestManager.shared.firstPendingRequest(for: origin, coinTypes: Array(WalletConstants.supportedCoinTypes)) {
  <...>
  } else if let request = WalletProviderPermissionRequestsManager.shared.pendingRequests(for: origin, coinType: .eth).first {
  <...>  
}
```

Reply:
sure thing. fixed at [5bfc01c](https://github.com/brave/brave-ios/pull/6075/commits/5bfc01c1dbe59d88299f4b2d3d51f6f53387901c)